dispatcher.go

```diff
@@ -213,3 +245,26 @@ func (d dispatcher) Stop() error {
 
 	return nil
 }
+
+// getUniqueOutbounds helps ensure we only start/stop the same outbound once
+func (d dispatcher) getUniqueOutbounds() []transport.Outbound {
+	var unique []transport.Outbound
```

Comment:
This needs verifying but you can *probably* use a `map[transport.Outbound]` to dedupe.